composer.json

```
@@ -35,5 +35,8 @@
     },
     "bin": [
         "parallel-lint"
-    ]
+    ],
+    "scripts": {
+        "tests": "tester -p php tests"
```

Suggestion:

```suggestion
        "tests": "@php vendor/bin/tester -p php tests"
```

* The `@php` will make sure that the PHP version used by Composer is used to run this script and not a system default version.
* The `vendor/bin` makes sure that the local version of `tester` is used and not a version potentially installed via `composer global` or that another script in the path called `tester` is called.

---

Nice, fixed & force-pushed!

---

did you mean `@composer exec tester -- -p php tests` ? - vendor is not always where composer has it ... .

---

Well, I didn't for sure.

IIRC, running it like you suggest would be problematic when there is also a (potentially different version) of `tester` installed in Composer global and I also very much doubt it will work on Windows.

Either way, running the tests is a dev contributor only thing, so if it works for the people currently contributing...

---

Yes, I'm also not totally sure about it honestly (and there are ugly problems with `composer exec` due that composer delegates to symfony console). @composer exec should not delegate to global, or at least it would be my understanding that this _should_ be only for @composer global exec. Otherwise I've learned already something new. In any case it should work as much on windows as `@php vendor/bin/tester` should, `$PHP_BINARY -f vendor/bin/tester` likely not.

And IIRC the original `tester -p php tests` is translated in a composer script automatically to the exec of the vendor (as configured) path. But would need to double-check composer sources again. There is something in it.

---

>  the original tester -p php tests is translated in a composer script automatically to the exec of the vendor (as configured) path.

Except that doesn't respect the PHP version used by Composer and will use the system default PHP version, which is not helpful.

And running with `@php vendor/bin/tester` does not work on Windows (well, it should, maybe, on the latest Composer as I remember seeing something in the changelog about it recently, but it didn't for the longest time and I haven't tested that type of script on the latest Composer yet, so can't actually be sure that it is fixed now on Windows).

---

Oh and `$PHP_BINARY` will also not work as that's a PHP 5.4 constant, while this library still supports PHP 5.3.

---

> Except that doesn't respect the PHP version used by Composer and will use the system default PHP version, which is not helpful.

Yes, you're right, `@composer exec` does not concern about the PHP version, just the path to the binary.  This was wishful thinking of me, I had not properly tested it first and was under a different impression. `@php ...` looks like the best variant then.

> Oh and $PHP_BINARY will also not work as that's a PHP 5.4 constant, while this library still supports PHP 5.3.

Actually it's an environment parameter set by Composer. It's effectively similar with the `@php` usage and similarly available when composer runs under PHP 5.3.

It is useful if you need the path to PHP in a script when its not at the very beginning (`@php` only works when it is at the very start of the script line).